### PR TITLE
zstd 1.5.5

### LINKED ISSUE #70: zstd version is out of date
### Comment:

Hi the included zstd version (1.5.2) was released on Jan 2022 and is more than a year older than the current stable version (1.5.5) released Apr 5 2023.

Is there a plan to update?

### PR DESCRIPTION
**Pull request by JohanMabille**
Replaces #69 
Fixes #70

### CODE REVIEW

**Review by hmaarrfk — APPROVED** [review]
I don't know too much about the speicfics of the windows package but looks good to me.

**Comment by JohanMabille** [comment]
By default, the symbols of a shared library are not exported on Windows, they are private to the shared lib. If you want to make them publicly visible, you usually declare them with `declspec(dllexport)` (and use them with `declspec(dllimport)` on the client side, thus the traditional XX_API macros in classes and functions declarations on Windows).

The problem here is that the symbols that are missing must not be publicly visible, so the only solution is to add the source file defining them to the executable which needs them.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   name: zstd-split
   version: {{ version }}
 
 source:
   url: https://github.com/facebook/zstd/archive/v{{ version }}.tar.gz
-  sha256: f7de13462f7a82c29ab865820149e778cbfe01087b3a55b5332707abf9db4a6e
+  sha256: 98e9c3d949d1b924e28e01eccb7deed865eefebf25c2f21c702e5cd5b63b85e1
   patches:
-    - patches/windows-cmake-pkg-config.patch  # [win]
+    - patches/cmake_windows.patch  # [win]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:11]: if you could just reference the PR here that would be great.

**Comment by hmaarrfk** [comment]
Great thanks. with the reference upstream at least there is somebody else knowledgeable in this particular library that has vetted the patch.

**Pull request merged**